File: ModernCocoaFarmerBindings/Bindings/ChildModifier.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows;
using System.Windows.Controls;
using System.Threading.Tasks;
using System.Xml.Serialization;
using BindingsKernel;

namespace MCF.Stats
{
	public class ChildModifier : Modifier
	{
		[Serialize, DisplayPriority(5)]
		[XmlAttribute("applies_to_all_children"), DisplayName("Applies To All Children")]
		public bool AppliesToAllChildren { get; set; } = false;
	}
}

File: ModernCocoaFarmer/Source/Stats/ChildModifier.cpp
#include "Stats/ChildModifier.h"
#include "UtilityHeaders/ScriptableObjectHeaders.h"


namespace MCF
{
  namespace Stats
  {
    REGISTER_SCRIPTABLE_OBJECT(ChildModifier);

    const std::string ChildModifier::APPLIES_TO_ALL_CHILDREN_FIELD_NAME = "applies_to_all_children";

    //------------------------------------------------------------------------------------------------
    ChildModifier::ChildModifier() :
      m_appliesToAllChildren(createValueField<bool>(APPLIES_TO_ALL_CHILDREN_FIELD_NAME, false))
    {
    }
  }
}